Several layouts did define global key types, so keys for other groups were broken.
Closes: #373692, #385970  Thanks Robert Marlow and Jan Willem Stumpel

Submitted upstream: https://bugs.freedesktop.org/show_bug.cgi?id=8121

Index: xkb-data/symbols/cd
===================================================================
--- xkb-data.orig/symbols/cd
+++ xkb-data/symbols/cd
@@ -8,7 +8,7 @@
 
     name[Group1]= "Congo, Democratic Republic of the";
 
-    key.type = "FOUR_LEVEL";
+    key.type[Group1] = "FOUR_LEVEL";
 
     // Alphanumeric section
     // TILDE, COMBINING TILDE
Index: xkb-data/symbols/gr
===================================================================
--- xkb-data.orig/symbols/gr
+++ xkb-data/symbols/gr
@@ -88,7 +88,7 @@
 
     name[Group1] = "Greece - Extended";
 
-    key.type = "THREE_LEVEL";
+    key.type[Group1] = "THREE_LEVEL";
 
     key <AE05> { [	5,		percent,	EuroSign	] };
 
@@ -108,7 +108,7 @@
 
     name[Group1] = "Greece - Polytonic";
 
-    key.type = "THREE_LEVEL";
+    key.type[Group1] = "THREE_LEVEL";
 
     key <AD11> { [ 	dead_tilde,	dead_diaeresis,	dead_macron	] };
     key <AD12> { [	dead_iota,	VoidSymbol,	dead_breve	] };
Index: xkb-data/symbols/il
===================================================================
--- xkb-data.orig/symbols/il
+++ xkb-data/symbols/il
@@ -48,7 +48,7 @@
     key <AB08> { [ hebrew_taw,	greater		]	};
     key <AB09> { [ hebrew_finalzade,	less	]	};
 
-    key.type = "THREE_LEVEL";
+    key.type[Group1] = "THREE_LEVEL";
     
     key <TLDE> { [ semicolon,	asciitilde,0x10005b0	]}; // Sheva
     key <AB10> { [ period,	question,  0x10005c3	]}; // Sof Pasuq
Index: xkb-data/symbols/jp
===================================================================
--- xkb-data.orig/symbols/jp
+++ xkb-data/symbols/jp
@@ -10,7 +10,7 @@
 
     // Alphanumeric section
     key <HZTG> {
-	type="PC_SYSRQ",
+	type[Group1]="PC_SYSRQ",
 	symbols[Group1]= [ Zenkaku_Hankaku, Kanji ]
     };
     key <AE01> { [ kana_NU		]	};
@@ -69,18 +69,18 @@
     key <NFER> { [ Muhenkan		]	};
 
     key <XFER> {
-	type="PC_SYSRQ",
+	type[Group1]="PC_SYSRQ",
 	symbols[Group1]= [ Henkan, Mode_switch ]
     };
     key <HKTG> {
-	type="PC_SYSRQ",
+	type[Group1]="PC_SYSRQ",
 	symbols[Group1]= [ Hiragana_Katakana, Romaji ]
     };
 
     // End alphanumeric section
 
     key <PRSC> {
-        type= "PC_SYSRQ",
+        type[Group1]= "PC_SYSRQ",
         symbols[Group1]= [ Print, Execute ]
     };
 };
@@ -92,7 +92,7 @@
     include "us"
     // Alphanumeric section
     key <HZTG> {
-	type="PC_SYSRQ",
+	type[Group1]="PC_SYSRQ",
 	symbols[Group1]= [ Zenkaku_Hankaku, Kanji ]
     };
 
@@ -121,16 +121,16 @@
     key <NFER> { [ Muhenkan		]	};
 
     key <XFER> {
-	type="PC_SYSRQ",
+	type[Group1]="PC_SYSRQ",
 	symbols[Group1]= [ Henkan, Mode_switch ]
     };
     key <HKTG> {
-	type="PC_SYSRQ",
+	type[Group1]="PC_SYSRQ",
 	symbols[Group1]= [ Hiragana_Katakana, Romaji ]
     };
 
     key <PRSC> {
-        type= "PC_SYSRQ",
+        type[Group1]= "PC_SYSRQ",
         symbols[Group1]= [ Print, Execute ]
     };
 };
Index: xkb-data/symbols/la
===================================================================
--- xkb-data.orig/symbols/la
+++ xkb-data/symbols/la
@@ -67,11 +67,11 @@
     key <AB10> { [ 0x1000e9d,       parenright	]};
 
 override key <LFSH> {
-	type="PC_SYSRQ",
+	type[Group1]="PC_SYSRQ",
 	[ 	 Shift_L,		ISO_First_Group		]
     };
 override key <RTSH> {
-	type="PC_SYSRQ",
+	type[Group1]="PC_SYSRQ",
 	[ 	Shift_R,		ISO_Last_Group 		]
     };
 };
Index: xkb-data/symbols/lk
===================================================================
--- xkb-data.orig/symbols/lk
+++ xkb-data/symbols/lk
@@ -36,7 +36,7 @@
   include "us"
 // it is default - no details in the name
   name[Group1] = "Sri Lanka";
-  key.type = "FOUR_LEVEL";
+  key.type[Group1] = "FOUR_LEVEL";
 
   // q - p
   key <AD01> { [ 0x01000d8d, 0x01000d8e, 0x01000dd8, 0x01000df2 ] };
Index: xkb-data/symbols/ru
===================================================================
--- xkb-data.orig/symbols/ru
+++ xkb-data/symbols/ru
@@ -165,7 +165,7 @@
     include "ru(winkeys)"
     name[Group1]= "Russia - Tatar";
 
-    key.type[group2]="FOUR_LEVEL";
+    key.type[group1]="FOUR_LEVEL";
 
     key <TLDE> {       [      0x010004bb,      0x010004ba,
                              Cyrillic_io,     Cyrillic_IO      ]       };
Index: xkb-data/symbols/ca
===================================================================
--- xkb-data.orig/symbols/ca
+++ xkb-data/symbols/ca
@@ -445,7 +445,7 @@
     key <AB10>	{ [    eacute,     Eacute,     dead_acute, NoSymbol,
     		   NoSymbol, dead_abovedot ]	};
 
-    key <SPCE>	{ type= "FOUR_LEVEL", [     space,      space,   nobreakspace, NoSymbol ]	};
+    key <SPCE>	{ type[Group1]= "FOUR_LEVEL", [     space,      space,   nobreakspace, NoSymbol ]	};
 
     modifier_map Control{ Control_L };
 
